components/category-select-table.vue

```diff
@@ -1,59 +1,57 @@
 <template>
 
-    <b-container fluid>
-
-        <!-- Heading for category select component -->
-        <b-row>
-            <h3>{{ title }}</h3>
-        </b-row>
-
-        <!-- Instructions prompting the user how to link categories and columns -->
-        <b-row>
-            <p class="instructions-text">
-                {{ instructions }}
-            </p>
-        </b-row>
+    <div>
 
         <!-- Category selection table -->
-        <b-row>
-            <b-table
-                outlined
-                selectable
-                head-variant="dark"
-                :items="categoryTable"
-                @row-selected="selectCategory"
-                select-mode="single"
-                selected-variant=""
-                :tbody-tr-class="styleTableRow"
-                thead-class="hidden" />
+        <b-row class="no-padding-row">
+
+            <b-col cols="12" class="no-padding-col">
+                <b-table
+                    outlined
+                    selectable
+                    head-variant="dark"
+                    :items="categoryTable"
+                    @row-selected="selectCategory"
+                    select-mode="single"
+                    selected-variant=""
+                    :tbody-tr-class="styleTableRow"
+                    thead-class="hidden" />
+            </b-col>
         </b-row>
 
-    </b-container>
+    </div>
 
 </template>
 
 <script>
 
-    export default {
-
-        props: {
+    // Allows for reference to store data by creating simple, implicit getters
+    import { mapGetters } from "vuex";
 
-            categories: { type: Array, required: true },
-            categoryClasses: { type: Object, required: true },
-            instructions: { type: String, required: true },
-            title: { type: String, required: true }
-        },
+    export default {
 
         data() {
 
             return {
 
-                selectedCategory: this.categories[0]
+                selectedCategory: ""
             };
         },
 
+        created() {
+
+            // Set the first given category as the selected one
+            this.selectedCategory = this.categories[0];
```

Comment:
We said the categorization page will keep the `selectedCategory` state for both the `category-select` and the `column-linking` component. But inside the `selectedCategory` component we are also keeping a locally scoped `selectedCategory` state variable. Does this mean we are keeping the `"selectedCategory"` state in two places then? 

There are two issues I see with this: 
1. To know what the `selectedCategory` will default to on the parent page, I need to go and look inside the defaults of the component. And then the component has to emit that default choice back to the parent (before any user interaction has happened), so that I can start directly with annotating the column-linking table. 
2. There is a risk of divergence if we're not careful. E.g.: we forget about this double-state and write a process that directly changes the `selectedCategory` state on the parent page (e.g. set it to category 2). Because the parent page doesn't communicate this to the `category-select` component, the component state might now be different (e.g. category 1). And now my UI (highlighted cat1) no longer matches my UX (I'm annotating cat2). 

Long text, short point: I think we should keep state only in one place and have the rest just respond to it! So I would suggest passing `selectedCategory` as a prop to this component and removing the local state.

Reply:
Just as a sidenote: I do not actually understand how the component emits its local choice of `selectedCategory` back to the parent without user input. In other words: I do not understand why in the current app version, I can begin annotating with `"Subject ID"` without clicking on the `category-select` table. I just realized that after trying to think about the data flow here.

Reply:
@surchs Thanks for the comments!

To address your first point. Thanks for taking the time to think this through, but I think this is one of those cases where if you see something in a PR that's odd/off, push it back to the submitter right away. In this case, I 100% agree with you. In fact, I'm not quite sure why `selectedCategory` is not being passed in as a prop. My guess is that I'm unused to working on components separate from their integration. I am currently on the categorization page refactor itself and yes, the categorization page is where this value will sit (until if/when we ever decide to move it to the store).

>I do not understand why in the current app version, I can begin annotating with "Subject ID" without clicking on the category-select table. I just realized that after trying to think about the data flow here.

The answer to this is in the `data` of the `category-select-table via `selectedCategory: this.categories[0]`. This might in fact be an inconsistent implementation that we have just overlooked where this field was sitting in two places but not causing any issues in behavior.

Reply:
Resolved with https://github.com/neurobagel/annotation_tool/pull/229/commits/56cab8c016e5c0c30f4d82fd84585210918f7985